src/libsync/propagateupload.h

```diff
@@ -290,23 +290,48 @@ class PropagateUploadFileNG : public PropagateUploadFileCommon {
     uint _transferId; /// transfer id (part of the url)
     int _currentChunk; /// Id of the next chunk that will be sent
     bool _removeJobError; /// If not null, there was an error removing the job
+    quint64 _lastChunkSize; /// current chunk size
```

Comment:
I am not fully sure how Bandwidth Manager works, I did not touch it. Of course, we can play with it and test what gives the enhancement easier.